stix2validator/shoulds.py

```diff
@@ -274,14 +274,16 @@ def relationships_strict(instance):
         return
 
     r_type = instance['relationship_type']
-    r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
-    r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    try:
+        r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
+        r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    except (AttributeError, TypeError):
```

Comment:
Yes, you got it - for instance if one of the two is set to null. However, I realize that if one of the two is null but the other is a valid id but just of a type invalid for relationships, then it still makes sense to give the error for the latter.